src/martialarts.cpp

```diff
@@ -348,7 +349,8 @@ bool ma_requirements::is_valid_player( const player &u ) const
         ( unarmed_allowed && u.unarmed_attack() &&
           ( !strictly_unarmed || !u.is_armed() ) ) ||
         ( is_valid_weapon( u.weapon ) &&
-          ( melee_allowed || u.style_selected.obj().has_weapon( u.weapon.typeId() ) ) );
+          // Below && used to be || - this change apparently makes techniques respect melee_allowed
+          ( melee_allowed && u.style_selected.obj().has_weapon( u.weapon.typeId() ) ) );
```

Comment:
This line was actually correct before, according to the comments above.  `melee_allowed` means "Any generic weapon is allowed"

Reply:
Okay, I understand. My problem then is with strictly_unarmed which is not being respected for martial weapons. I'll look into fixing this as it appears to be the actual bug.